### PR TITLE
Suparna_Assignment_31_15Min

### PR DESCRIPTION
**Pull request by SuparnaNikam**
ExceptionHandling

### CODE REVIEW

File: src/suparnaNJune20/Assignment_31/InputNumberExceptionHandling.java [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+				scanner.close();
+		}
+	
+	public static void main(String[] args) {
+		// TODO Auto-generated method stub
+		Scanner scanner = new Scanner(System.in);
+		System.out.println("Enter total number of count of number  to find sum");
+new InputNumberExceptionHandling().sumOfNumber(scanner.nextInt());
+	}
+
```
Comment (VireshJJun20) [inline src/suparnaNJune20/Assignment_31/InputNumberExceptionHandling.java]: extra spacing

File: src/suparnaNJune20/Assignment_31/InputNumberExceptionHandling.java [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+		for ( cnt = 1; cnt <= inputCnt; cnt++) {
+			
+			try {
+				System.out.println("Enter "+ cnt +" input Number ");
+				sum += scanner.nextInt();
+				
+			} catch (InputMismatchException inputMismatchException) {
+				System.out.println("Enter Number input not a string  ");
+				scanner.next();
+
```
Comment (VireshJJun20) [inline src/suparnaNJune20/Assignment_31/InputNumberExceptionHandling.java]: extra spacing

File: src/suparnaNJune20/Assignment_31/InputNumberExceptionHandling.java [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+	{
+		
+		
+		Scanner scanner = new Scanner(System.in);
+		for ( cnt = 1; cnt <= inputCnt; cnt++) {
+			
+			try {
+				System.out.println("Enter "+ cnt +" input Number ");
+				sum += scanner.nextInt();
+
```
Comment (VireshJJun20) [inline src/suparnaNJune20/Assignment_31/InputNumberExceptionHandling.java]: extra spacing

File: src/suparnaNJune20/Assignment_31/InputNumberExceptionHandling.java [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+import java.util.Scanner;
+public class InputNumberExceptionHandling {
+	static int sum = 0, cnt = 0;
+	public void sumOfNumber(int inputCnt)
+	{
+		
+		
+		Scanner scanner = new Scanner(System.in);
+		for ( cnt = 1; cnt <= inputCnt; cnt++) {
+
```
Comment (VireshJJun20) [inline src/suparnaNJune20/Assignment_31/InputNumberExceptionHandling.java]: extra spacing

File: src/suparnaNJune20/Assignment_31/InputNumberExceptionHandling.java [hunk 5]
```diff
@@ -0,0 +1,34 @@
+package suparnaNJune20.Assignment_31;
+import java.util.InputMismatchException;
+import java.util.Scanner;
+public class InputNumberExceptionHandling {
+	static int sum = 0, cnt = 0;
+	public void sumOfNumber(int inputCnt)
+	{
+
```
Comment (VireshJJun20) [inline src/suparnaNJune20/Assignment_31/InputNumberExceptionHandling.java]: Extra spacing, please remove it.

File: src/suparnaNJune20/Assignment_31/InputNumberExceptionHandling.java [hunk 6]
```diff
@@ -0,0 +1,34 @@
+package suparnaNJune20.Assignment_31;
+import java.util.InputMismatchException;
+import java.util.Scanner;
+public class InputNumberExceptionHandling {
+	static int sum = 0, cnt = 0;
```
Comment (VireshJJun20) [inline src/suparnaNJune20/Assignment_31/InputNumberExceptionHandling.java]: No need to declare the variables as static.

**Review by VireshJJun20 — CHANGES_REQUESTED** [review]
Please check my comments and update where ever needed. Thanks.

File: src/suparnaNJune20/Assignment_31/InputNumberExceptionHandling.java [hunk 7]
```diff
@@ -0,0 +1,34 @@
...
+				System.out.println("Enter "+ cnt +" input Number ");
+				sum += scanner.nextInt();
+				
+			} catch (InputMismatchException inputMismatchException) {
+				System.out.println("Enter Number input not a string  ");
+				scanner.next();
+				
+			}
+		}
+		System.out.println("Sum = "+sum);
```
Comment (VireshJJun20) [inline src/suparnaNJune20/Assignment_31/InputNumberExceptionHandling.java]: We need to display all the given input numbers and the sum in single line.
eg: "Values are 10, 30, 40 and total is 80."

File: src/suparnaNJune20/Assignment_31/InputNumberExceptionHandling.java (hunk 1 above)
Comment (SuparnaNikam) [reply src/suparnaNJune20/Assignment_31/InputNumberExceptionHandling.java]: Changes Done.

File: src/suparnaNJune20/Assignment_31/InputNumberExceptionHandling.java (hunk 2 above)
Comment (SuparnaNikam) [reply src/suparnaNJune20/Assignment_31/InputNumberExceptionHandling.java]: Changes Done.

File: src/suparnaNJune20/Assignment_31/InputNumberExceptionHandling.java (hunk 3 above)
Comment (SuparnaNikam) [reply src/suparnaNJune20/Assignment_31/InputNumberExceptionHandling.java]: Changes Done.

File: src/suparnaNJune20/Assignment_31/InputNumberExceptionHandling.java (hunk 4 above)
Comment (SuparnaNikam) [reply src/suparnaNJune20/Assignment_31/InputNumberExceptionHandling.java]: Changes Done.

File: src/suparnaNJune20/Assignment_31/InputNumberExceptionHandling.java (hunk 5 above)
Comment (SuparnaNikam) [reply src/suparnaNJune20/Assignment_31/InputNumberExceptionHandling.java]: Changes Done.

File: src/suparnaNJune20/Assignment_31/InputNumberExceptionHandling.java (hunk 6 above)
Comment (SuparnaNikam) [reply src/suparnaNJune20/Assignment_31/InputNumberExceptionHandling.java]: Changes Done.

File: src/suparnaNJune20/Assignment_31/InputNumberExceptionHandling.java (hunk 7 above)
Comment (SuparnaNikam) [reply src/suparnaNJune20/Assignment_31/InputNumberExceptionHandling.java]: Changes Done.

File: src/suparnaNJune20/Assignment_31/InputNumberExceptionHandling.java [hunk 8]
```diff
@@ -1,35 +1,40 @@
...
+				sum += inputNo;
+				inputStr = inputStr.append(inputNo);
+				inputStr = inputStr.append(" , ");
+
 			} catch (InputMismatchException inputMismatchException) {
 				System.out.println("Enter Number input not a string  ");
 				cnt--;
-				scanner.next();
-				
+				inputStr = inputStr.append(scanner.next());
```
Comment (VireshJJun20) [inline src/suparnaNJune20/Assignment_31/InputNumberExceptionHandling.java:24]: Here, if we provide invalid input (eg: abc) then inputStr still contains that invalid input. then it gets appended with the next input given.

**Review by VireshJJun20 — CHANGES_REQUESTED** [review]
Please check my comments and update accordingly.

**Comment by KrishnaKTechnocredits** [comment]
@SuparnaNJune20 You should work on Viresh 's all commnets.

**Pull request merged**